rest_framework_api_key/models.py

```diff
@@ -23,10 +24,18 @@ def is_valid(self, key: str) -> bool:
         prefix, _, _ = key.partition(".")
 
         try:
-            obj = self.get(id__startswith=prefix, revoked=False)
+            obj = self.filter(id__startswith=prefix, revoked=False).annotate(
+                is_expiry=Case(
+                    When(expiry_date__isnull=True, then=V(False)),
+                    When(expiry_date__gte=Now(), then=V(False)),
+                    default=V(True),
+                    output_field=BooleanField()
+                )
+            ).filter(is_expiry=False).first()
```

Comment:
Wow, this is a clever one!

While I like it, wouldn't it be simpler (and faster?) to get the key and validate it first (just as before), and *then* check its expiry date?

This way we definitely know that the key gets rejected *because* it has expired.

I believe this could be implemented using a `has_expired` property on `APIKey`.